### PR TITLE
Ensure there is a dvc.yaml file when running experiment

### PR DESCRIPTION
**Pull request by sroy3**
Part of #3058 

Creates an empty (seems to be all that is needed) dvc.yaml file when none exists before running experiment.

https://user-images.githubusercontent.com/3683420/215572713-4700089d-11c1-40a3-99fc-66d38be297b1.mov

Should we add a notification saying we created the file? Or even ask the user if they would like to create the file? Seems low impact enough to me to simply create the file for them.

### CODE REVIEW

**Comment by mattseddon** [comment]
Talked through ideas for the next step with @sroy3. 

## Ideas that came up:

### Populating dvc.yaml
Run button shows a modal which asks the user if they want to add a dvc.yaml so they run an experiment.
Quick input then asks for the relative path to their training script.
dvc.yaml is created containing: 
```
stages:
  train:
    cmd: python relative/path/to/train/script
```

### Final step of onboarding
Add another step to the end of onboarding which shows the location of the run button (editor/title) and details what clicking it will do (see above idea).

File: extension/src/experiments/workspace.ts [hunk 1]
```diff
@@ -213,12 +213,19 @@ export class WorkspaceExperiments extends BaseWorkspaceWebviews<
...
+    commandId: CommandId,
+    ensureDvcYamlFileExists?: boolean
+  ) {
     const cwd = await this.getFocusedOrOnlyOrPickProject()
     if (!cwd) {
       return
     }
 
+    if (ensureDvcYamlFileExists) {
+      ensureOrCreateDvcYamlFile(cwd)
```
Comment (mattseddon) [inline extension/src/experiments/workspace.ts:226]: [Q] `findOrCreateDvcYamlFile`?

Comment (mattseddon) [reply extension/src/experiments/workspace.ts:226]: or just `ensureDvcYamlFile`?

**Review by mattseddon — APPROVED** [review]
Couple things to check

1. does running an experiment with an empty `dvc.yaml` still respect the params created by Dvclive?
2. does this create any other weirdness when running experiments?

**Review by julieg18 — APPROVED** [review]

**Comment by sroy3** [comment]
> Couple things to check
> 
> 1. does running an experiment with an empty `dvc.yaml` still respect the params created by Dvclive?
> 2. does this create any other weirdness when running experiments?

This works great for already set up projects (like https://github.com/iterative/dvc-get-started-cv/tree/1-notebook-dvclive), but it seems like it needs more than that for newly created projects. With the `dvc.yaml` file, it'll run, but the table will still have no columns.

**Comment by sroy3** [comment]
Closing this as this is too simple as a first step. Will add a prompt to the training script and re-open.

**Pull request closed without merging**